pkg/resources/resources.go

```diff
@@ -149,62 +152,72 @@ type NodeSchedulingMetadata struct {
 // Zero returns a Resources object with quantities of zero
 func Zero() *Resources {
 	return &Resources{
-		CPU:    *resource.NewQuantity(0, resource.DecimalSI),
-		Memory: *resource.NewQuantity(0, resource.BinarySI),
+		CPU:       *resource.NewQuantity(0, resource.DecimalSI),
+		Memory:    *resource.NewQuantity(0, resource.BinarySI),
+		NvidiaGPU: *resource.NewQuantity(0, resource.DecimalSI),
 	}
 }
 
 //AddFromReservation modifies the receiver in place.
-func (r *Resources) AddFromReservation(reservation *v1beta1.Reservation) {
-	r.CPU.Add(reservation.CPU)
-	r.Memory.Add(reservation.Memory)
+func (r *Resources) AddFromReservation(reservation *v1beta2.Reservation) {
+	r.CPU.Add(*reservation.Resources.CPU())
+	r.Memory.Add(*reservation.Resources.Memory())
+	r.NvidiaGPU.Add(*reservation.Resources.NvidiaGPU())
 }
 
 // Copy returns a clone of the Resources object
 func (r *Resources) Copy() *Resources {
 	return &Resources{
-		CPU:    r.CPU.DeepCopy(),
-		Memory: r.Memory.DeepCopy(),
+		CPU:       r.CPU.DeepCopy(),
+		Memory:    r.Memory.DeepCopy(),
+		NvidiaGPU: r.NvidiaGPU.DeepCopy(),
 	}
 }
 
 //Add modifies the receiver in place.
 func (r *Resources) Add(other *Resources) {
 	r.CPU.Add(other.CPU)
 	r.Memory.Add(other.Memory)
+	r.NvidiaGPU.Add(other.NvidiaGPU)
 }
 
 //Sub modifies the receiver in place
 func (r *Resources) Sub(other *Resources) {
 	r.CPU.Sub(other.CPU)
 	r.Memory.Sub(other.Memory)
+	r.NvidiaGPU.Sub(other.NvidiaGPU)
 }
 
 // AddFromResourceList modified the receiver in place
 func (r *Resources) AddFromResourceList(resourceList corev1.ResourceList) {
 	r.CPU.Add(resourceList[corev1.ResourceCPU])
 	r.Memory.Add(resourceList[corev1.ResourceMemory])
+	r.NvidiaGPU.Add(resourceList[v1beta2.ResourceNvidiaGPU])
```

Comment:
nit -- not too opinionated, and I know we would get the zero value of the Quantity if GPUs are not set, but I wonder if we want to have an explicit check for the sake of making it clear in code that it is not necessarily there.

Reply:
I feel like it would clutter the code quite a bit as in most functions we are doing something along these lines. I'd prefer to leave this out unless you think it adds a lot.